### PR TITLE
Retry on the minting in case of errors

### LINKED ISSUE #3547: Retry on the minting in case of errors
Once users mint and for whatever reasons drops connections, we should handle it in a graceful way and retry the connection.
Probably we should handle RPC errors with some retry?

In my case, retry worked minting on the second try.

### Handle retry for
- pinning
- on-chain transaction

image

image

### PR DESCRIPTION
**Pull request by Jarsen136**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
👇 \_ Let's make a quick check before the contribution.

### PR type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

### What's new?

- [x] PR closes #3547
- [ ] Requires deployment <rubick/magick_issue>
- [ ] <brief_description_of_what_I've_added>

### Before submitting Pull Request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

### Optional

- [ ] I've tested it at </rmrk/collection/26902bc2f7c20c546a-1FVG7>
- [ ] I've tested PR on mobile and everything seems works
- [ ] I found edge cases
- [ ] I've written some unit tests 🧪

### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG)

### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

I emit a mock error while minting, and it could retry to mint again successfully.
Kapture 2022-07-24 at 16 18 03

### CODE REVIEW

File: components/bsx/Create/CreateToken.vue [hunk 1]
```diff
@@ -313,6 +313,11 @@ export default class CreateToken extends mixins(
       if (e instanceof Error) {
         showNotification(e.toString(), notificationTypes.danger)
         this.stopLoader()
+
+        if (retryCount < 3) {
+          showNotification('You will retry to mint NFT')
```
Comment (kkukelka) [inline components/bsx/Create/CreateToken.vue:318]: sorry for being picky here but can we rephrase that
```suggestion
          showNotification('Retrying to complete minting process.')
```

File: components/bsx/Create/CreateToken.vue [hunk 2]
```diff
@@ -54,7 +54,7 @@
           label="mint.submit"
           :disabled="disabled"
           :loading="isLoading"
-          @click="submit" />
+          @click="() => submit()" />
```
Comment (kkukelka) [inline components/bsx/Create/CreateToken.vue:57]: i believe you can just call `submit()` here without arrow function
```suggestion
          @click="submit()" />
```

File: components/bsx/Create/CreateToken.vue [hunk 3]
```diff
@@ -261,7 +261,7 @@ export default class CreateToken extends mixins(
     return !this.listed || (price > 0 && price <= this.maxPrice)
   }
 
-  protected async submit(): Promise<void> {
+  protected async submit(retryCount = 0): Promise<void> {
```
Comment (kkukelka) [inline components/bsx/Create/CreateToken.vue:264]: why do we actually need this parameter here?

**Review by kkukelka — CHANGES_REQUESTED** [review]

File: components/bsx/Create/CreateToken.vue (hunk 3 above)
Comment (Jarsen136) [reply components/bsx/Create/CreateToken.vue:264]: Because it needs an initial value to limit the retry time. Or it will retry forever

Comment (Jarsen136) [reply components/bsx/Create/CreateToken.vue:264]: I limited it to three retries

Comment (kkukelka) [reply components/bsx/Create/CreateToken.vue:264]: Ahh whoops, didn't notice that the function is calling itself there.

**Review by kkukelka — APPROVED** [review]
all good now

**Comment by DeluneClaire** [comment]
I tried to mint NFT, but it all worked without error. Is there any way to mint with an error so I can try the retry?

**Comment by Jarsen136** [comment]
> I tried to mint NFT, but it all worked without error. Is there any way to mint with an error so I can try the retry?

It seems hard to make an error without running code. : )

**Comment by DeluneClaire** [comment]
> > I tried to mint NFT, but it all worked without error. Is there any way to mint with an error so I can try the retry?
> 
> It seems hard to make an error without running code. : )

I don't understand, what do you mean by that?

**Comment by Jarsen136** [comment]
> you

I mean It's so hard to find a way to mint with an error : )

**Review by DeluneClaire — APPROVED** [review]
works for me

**Comment by DeluneClaire** [comment]
> > I don't understand, what do you mean by that?
> 
> I mean It's so hard to find a way to mint with an error : )

Yeah, because you do a great job! :D from the beginning, there were many errors..

**Review by kkukelka — APPROVED** [review]

**Comment by Jarsen136** [comment]
@kkukelka Hey, seem to forget to pay the bounty. 👀

**Comment by kkukelka** [comment]
pay 100

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $100 @ 56.93 USD/KSM ~ 1.757 $KSM
🧗 [Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG](https://kusama.subscan.io/account/Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG)
🔗 [0xb7fad0b40a733ecdf4de4cb281de076cd27eca60970ebdc2a81b2379f8967e11](https://kusama.subscan.io/extrinsic/0xb7fad0b40a733ecdf4de4cb281de076cd27eca60970ebdc2a81b2379f8967e11)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Comment by kkukelka** [comment]
> @kkukelka Hey, seem to forget to pay the bounty. 👀

all good just wanted to wait till its merged
